src/inmanta/logging.py

```diff
@@ -99,15 +117,103 @@ def __init__(self, stream: TextIO = sys.stdout) -> None:
         :param stream: The stream to send log messages to. Default is standard output (sys.stdout).
         """
         self._options_applied = False
+        self._executing_compile_or_export_command = False
         self._handler: logging.Handler = logging.StreamHandler(stream=stream)
         self.set_log_level("INFO")
         formatter = self._get_log_formatter_for_stream_handler(timed=False)
         self.set_log_formatter(formatter)
 
+        self._logger_mode = LoggerMode.OTHER
+        self._default_log_level_factory = logging.getLogRecordFactory()
+        logging.setLogRecordFactory(self.custom_log_record_factory)
+        self._source_file_to_module_cache: dict[str, Optional[str]] = {}
+        # A cache for the `_get_module_name_for_source_file()` method.
+
         logging.root.handlers = []
         logging.root.addHandler(self._handler)
         logging.root.setLevel(0)
 
+    def _get_module_name_for_source_file(self, path_source_file: str) -> Optional[str]:
+        """
+        Return the module name that the given `path_source_file` belongs to or None if `path_source_file`
+        doesn't belong to a module.
+        """
+        if path_source_file in self._source_file_to_module_cache:
+            return self._source_file_to_module_cache[path_source_file]
+        dirs_containing_modules = []
```

Comment:
couldn't this one be cached as well? but I think it's overkill

Reply:
Good point. Initially I though we couldn't, but since the compiler doesn't install modules anymore, we can safely assume that the content of `dirs_containing_modules` will not change during execution.